mitodl/open-discussions
# Allow anonymous access for channels

#### What are the relevant tickets?
Fixes #600 

#### What's this PR do?
Adds a feature flag for anonymous access and support in `Api`. Updates permission classes to allow anonymous readonly access to channels.

#### How should this be manually tested?
 - Set `FEATURE_ANONYMOUS_ACCESS=True` in your `.env`
 - In a private session go to these endpoints and verify that you don't get an error: `/api/v0/channels/`, `/api/v0/channels/<some_public_channel_name>/`. `/api/v0/channels/` should have an empty list.
